### PR TITLE
First draft of initial prototype

### PR DESCRIPTION
**Pull request by SarahAlidoost**
relates #4

### CODE REVIEW

File: springtime.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    workflow["observations_test"], workflow["observations_train"] = split_data(
+        workflow["observations"], options["train_test_strategy"]
+        )
+    
+    workflow["metric_func"] = define_metrics(options["metric_name"])
+    workflow["model"] = select_model(options["model_name"])
+
+    workflow["name"] = name
+    workflow["options"] = options
```
Comment (Peter9192) [inline springtime.py:134]: It has quite some similarities to existing (but overkill complexity) tools, e.g. https://www.mlflow.org/docs/latest/quickstart.html#using-the-tracking-api

Notice the analogy between how you "save workflow settings", and they "log metrics"

File: springtime.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    return model, Y_predict, metric_val
+
+
+def run_sklearn_workflow(workflow):
+    model = workflow["model"]
+    
+    # organize data for fitting
+    Y, X, _ = models_utils.misc.temperature_only_data_prep(
+        workflow["observations_train"], workflow["predictors"], for_prediction=False
+        )
```
Comment (Peter9192) [inline springtime.py:162]: Once we have our own sample data, the sklearn workflow should no longer have to depend on the same preprocessing as py-phenology.

**Review by Peter9192 — APPROVED** [review]
Very neat PR! I want to see investigate more how it aligns with existing workflow/orchestration packages, perhaps there is something we can reuse. Then, we should have a good brainstorm about further designing this.

File: basic_workflow_example.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Required dependencies for pyPhenology:
+Python
+scipy
+numpy
+pandas
+joblib
+
+pyPhenology
+scikit-learn #for ML methods
+seaborn #for plotting
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply basic_workflow_example.py]: Done!

File: basic_workflow_example.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+all_results = {
+    "dataset": [],
+    "phenophase": [],
+    "model": [],
+    "rmse": [],
+}
+
+for phenophase in phenophases:
+    for model_name in models_to_test:
+        ## Create options for workflow
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply basic_workflow_example.py]: Done!

**Comment by SarahAlidoost** [comment]
> Very neat PR! I want to see investigate more how it aligns with existing workflow/orchestration packages, perhaps there is something we can reuse. Then, we should have a good brainstorm about further designing this.

thanks for reviewing. Your comments are addressed. I will merge the PR :)

**Pull request merged**